### PR TITLE
[GraphBolt] Set persistent_workers in MultiProcessDataLoader

### PR DESCRIPTION
**Pull request by RamonZhou**
## Description
Given that the starting of workers is slow, we can set `persistent_workers` to True to avoid restarting at each epoch.
The multigpu example is also modified correspondingly.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/dataloader.py [hunk 1]
```diff
@@ -78,14 +78,21 @@ class MultiprocessingWrapper(dp.iter.IterDataPipe):
...
         The number of worker processes. Default is 0, meaning that there
         will be no multiprocessing.
+    persistent_workers : bool, optional
+        If True, the data loader will not shut down the worker processes after a
+        dataset has been consumed once. This allows to maintain the workers
+        instances alive. If None, it will default to True when num_workers > 0.
     """
 
-    def __init__(self, datapipe, num_workers=0):
+    def __init__(self, datapipe, num_workers=0, persistent_workers=None):
```
Comment (frozenbugs) [inline python/dgl/graphbolt/dataloader.py:87]: 1. optional boolean is weird. 2. It is weird to make None as default True.

Just set the default to True, and `persistent_workers = (num_workers > 0) and persistent_workers` should be enough.

File: examples/multigpu/graphbolt/node_classification.py [hunk 2]
```diff
@@ -148,20 +148,10 @@ def create_dataloader(
 
 
 @torch.no_grad()
-def evaluate(rank, args, model, graph, features, itemset, num_classes, device):
+def evaluate(rank, model, dataloader, num_classes, device):
```
Comment (peizhou001) [inline examples/multigpu/graphbolt/node_classification.py:151]: Is this change related to peresistent worker? if not, move this to another PR.

**Review by peizhou001 — APPROVED** [review]
LGTM except the left one comment.

File: examples/multigpu/graphbolt/node_classification.py (hunk 2 above)
Comment (RamonZhou) [reply examples/multigpu/graphbolt/node_classification.py]: Okay

**Review by frozenbugs — APPROVED** [review]
